{
  control.panic(cause);
  if (rethrow) {
    throw Exceptions.launderedException(cause);
  }
}
